### PR TITLE
Fix missing bottom and right borders in table cells (Issue #160)

### LINKED ISSUE #160: Bottom border missing in table
I'm trying to convert an HTML table like this one, but the border of at the bottom of the table is missing.
Image

Code is:
```
await HTMLtoDOCX(
    "<table><tr><td>left</td><td>right</td></tr></table>",
    undefined,
    { table: { borderOptions: { size: 1, stroke: "single", color: "000000" } } });
```
I'm not entirely sure how to interpret the content of the `word/document.xml` file that is generated (see full file: [document.xml](https://github.com/user-attachments/files/24238937/document.xml)), but I noticed those tags:
```
<w:tcBorders>
  <w:top w:val="single" w:sz="1" w:space="0" w:color="000000"/>
  <w:left w:val="single" w:sz="1" w:space="0" w:color="000000"/>
</w:tcBorders>
```
and
```
<w:tcBorders>
  <w:top w:val="single" w:sz="1" w:space="0" w:color="000000"/>
  <w:left w:val="single" w:sz="1" w:space="0" w:color="000000"/>
  <w:right w:val="single" w:sz="1" w:space="0" w:color="000000"/>
</w:tcBorders>
```
It's as if a `<w:bottom>` tag should have been there, too.

Did I miss something?

(PS: thanks for taking over the old https://github.com/privateOmega/html-to-docx library; this is much appreciated!)

### PR DESCRIPTION
**Pull request by nicolasiscoding**
## Summary

Fixes #160 - Missing bottom and right borders in table cells when using `borderOptions` configuration.

## Problem

When converting HTML tables to DOCX with `borderOptions` specified, bottom and right borders were missing from table cells. Only top and left borders were being rendered.

## Root Causes

1. **`buildTableCellBorders()` function bug**: The function was using the wrong variable reference after destructuring, causing border properties to be lost during iteration.

2. **`buildTableCell()` edge-only border application**: Table-level borders were only being applied to edge cells (first/last row/column), leaving middle cells without any borders.

### Changes Made

1. **Fixed `buildTableCellBorders()` (lines 1766, 1769)**:
   - Changed to use destructured `borders` variable instead of original `tableCellBorder` object
   - This matches the pattern already used in `buildTableBorders()`

2. **Fixed `buildTableCell()` (lines 2572-2594)**:
   - Modified to apply table-level borders to ALL cells, not just edge cells
   - This ensures complete grid rendering when `borderOptions` is specified

### Test Coverage

Added comprehensive test suite with 11 tests in `tests/table-cell-borders.test.js`:

- ✅ Basic border rendering with borderOptions
- ✅ All four borders (top, bottom, left, right)
- ✅ Different stroke styles (dashed, dotted, double, single)
- ✅ Custom colors
- ✅ Complex tables (multi-row, with headers)
- ✅ Edge cases (single cell, hidden borders)

### Before Fix
- 10/11 tests failing
- Bottom and right borders missing in generated XML

### After Fix
- **All 341 tests passing** (11 new + 330 existing)
- Bottom and right borders correctly present in all cells

## Development Approach

This fix was implemented using **Test-Driven Development (TDD)**:
1. **RED**: Wrote failing tests demonstrating the bug
2. **GREEN**: Implemented minimal fix to make tests pass
3. **REFACTOR**: Cleaned up code while keeping tests green

## Example

**Before Fix:**
```xml
<w:tcBorders>
  <w:top w:val="single" w:sz="1" w:space="0" w:color="000000"/>
  <w:left w:val="single" w:sz="1" w:space="0" w:color="000000"/>
</w:tcBorders>
```

**After Fix:**
```xml
<w:tcBorders>
  <w:top w:val="single" w:sz="1" w:space="0" w:color="000000"/>
  <w:bottom w:val="single" w:sz="1" w:space="0" w:color="000000"/>
  <w:left w:val="single" w:sz="1" w:space="0" w:color="000000"/>
  <w:right w:val="single" w:sz="1" w:space="0" w:color="000000"/>
</w:tcBorders>
```

## Checklist

- [x] Tests added and passing
- [x] No regressions (all existing tests pass)
- [x] Code follows existing patterns
- [x] Issue linked and will auto-close on merge

### CODE REVIEW

File: tests/table-cell-borders.test.js [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+      // After fix: should contain all four borders including <w:bottom>
+      const tcBordersRegex = /<w:tcBorders>(.*?)<\/w:tcBorders>/gs;
+      const tcBordersMatches = parsed.xml.match(tcBordersRegex);
+
+      expect(tcBordersMatches).not.toBeNull();
+      expect(tcBordersMatches.length).toBeGreaterThan(0);
+
+      // Verify bottom border exists as reported missing in the issue
+      const bottomBorderRegex = /<w:bottom\s+w:val="single"\s+w:sz="1"\s+w:space="0"\s+w:color="000000"\s*\/>/;
+      const hasBottomBorder = tcBordersMatches.some((tcBorders) =>
```
Comment (dt-eric-lefevreardant) [inline tests/table-cell-borders.test.js]: If you recall #160, I mentioned that there were _two_ `<w:tcBorders>` tags that were missing the `<w:bottom>` sub-tag. I think it's worth testing them all.
```suggestion
      const hasBottomBorder = tcBordersMatches.every((tcBorders) =>
```
Maybe it's worth making this clear in the expectation above, replacing
```
expect(tcBordersMatches.length).toBeGreaterThan(0);
```
with
```
expect(tcBordersMatches).toHaveLength(2);
```

**Review by dt-eric-lefevreardant — APPROVED** [review]

File: tests/table-cell-borders.test.js (hunk 1 above)
Comment (nicolasiscoding) [reply tests/table-cell-borders.test.js]: Thanks for this!

**Review by K-Kumar-01 — CHANGES_REQUESTED** [review]
Breaking the existing functionality. Needs changes.

**Comment by nicolasiscoding** [comment]
Closing — Issue #160 is already resolved on \`develop\` after the table alignment + border fixes shipped in #186 (and predecessors).

Verified just now with the exact reproducer from the issue:
\`\`\`js
await HTMLtoDOCX(
  "<table><tr><td>left</td><td>right</td></tr></table>",
  undefined,
  { table: { borderOptions: { size: 1, stroke: "single", color: "000000" } } }
);
\`\`\`
Both cells now emit all four \`<w:top>\`, \`<w:bottom>\`, \`<w:left>\`, \`<w:right>\` elements in their \`<w:tcBorders>\`.

Kushal's review feedback (#162#issuecomment) flagged that this branch's changes broke two scenarios (\`border-style: hidden\` + \`border="1"\`, and per-side mixed-color/width tables). Since the underlying bug is already fixed elsewhere, and rebasing this branch onto current \`develop\` would also revert today's merges (page-break support, strikethrough handling, blockquote image context fix, table alignment, fixupTableCellBorder swap fix), it's cleaner to close this and reopen a targeted PR if any residual border issue surfaces.

Closes #160 functionally via #186.
